pkg/ccrm/ccrm_exec.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2024 EdgeXR, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ccrm
+
+import (
+	"context"
+
+	"github.com/edgexr/edge-cloud-platform/api/edgeproto"
+	"github.com/edgexr/edge-cloud-platform/pkg/log"
+	"github.com/opentracing/opentracing-go"
+)
+
+func (s *CCRMHandler) ProcessExecRequest(ctx context.Context, in *edgeproto.CloudletExecReq) (*edgeproto.ExecRequest, error) {
+	_, platform, err := s.getCloudletPlatform(ctx, in.CloudletKey)
+	if err != nil {
+		return nil, err
+	}
+	resp := make(chan *edgeproto.ExecRequest, 1)
+
+	go func() {
```

Comment:
What's a point of spawning a go routine, if we just sit on the channel recv waiting for that function to finish?

Reply:
Good point, I probably just copied it from the crm code. I'll unwrap it.

Reply:
Actually Lev, we do actually need the go routine. The ProcessExecReq() function needs to send back the response before it completes, and then block to keep the pipes open to read/write from the shell process (for an interactive shell). Once the shell exits, then the ProcessExecReq() exists. Alternatiavely, ProcessExecReq may fail before running the interactive shell, in which case it doesn't block, and we send back the failure. I'll add some comments.